_utils/firefox_profile.py

```diff
@@ -0,0 +1,96 @@
+"""
+This module uses Fabric API to generate a Firefox Profile by concatenating the
+following preferences files:
+- ./_utils/prefs.ini
+- ./<application>/prefs.ini
+- ./<application>/<test_type>/prefs.ini
+
+The profile is then created using the specified name and saved to the ../_temp/
+directory.
+"""
+
+try:
+    import configparser  # Python 3
+except:
+    import ConfigParser as configparser  # Python 2
+
+import os
+import shutil
+
+import configargparse
+from fabric.api import local  # It looks like Fabric may only support Python 2.
+
+PATH_PROJECT = os.path.abspath('../')
+PATH_TEMP = os.path.join(PATH_PROJECT, '_temp')
+FILE_PREFS = 'prefs.ini'
+
+config = configparser.ConfigParser()
+
+
+def _parse_args():
+    """Parses out args for CLI"""
+    parser = configargparse.ArgumentParser(
+        description='CLI tool for creating Firefox profiles via mozprofile CLI')
+    parser.add_argument('-a', '--application',
+                        required=True,
+                        help='Application to test. Example: "loop-server"')
+    parser.add_argument('-t', '--test-type',
+                        required=True,
+                        help='Application test type. Example: "stack-check"')
+    parser.add_argument('-e', '--env',
+                        help='Test environment. Example: "dev", "stage", ...')
+    parser.add_argument('-p', '--profile',
+                        required=True,
+                        help='Profile name.')
+
+    args = parser.parse_args()
+    return args, parser
+
+
+def prefs_paths(application, test_type, env='stage'):
+    path_global = os.path.join(PATH_PROJECT, '_utils', FILE_PREFS)
+    path_app_dir = os.path.join(PATH_PROJECT, application)
+    path_app = os.path.join(path_app_dir, FILE_PREFS)
+    path_app_test_type = os.path.join(path_app_dir, test_type, FILE_PREFS)
+
+    valid_paths = [path_global]
+
+    if os.path.exists(path_app):
+        config.read(path_app)
+        # Make sure the specified INI file has the specified section.
+        if config.has_section(env):
```

Comment:
@rpappalax I broke down and added that extra check we were talking about. Now I check that:
1. The specified application (ie: "loop-server") has a "prefs.ini" file
2. The "{application}/prefs.ini" file has a "[{env}]" section.

If (1) and (2) are happy above, we can append the valid path to the valid paths list.

Previously it would have just failed if you specified "[dev]" but the INI didn't have a "dev" section.

Ditto with the check on L66 below. :kiss: